code/reproduction/similarity.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import print_function
+from __future__ import division
+import nibabel as nib
+import numpy as np
+from scipy import stats
+from multiprocessing import Process
+import itertools
+import sys
+from os import path
+import sharedmem as sm
+import gc
+
+REPO_HOME_RELATIVE_PATH = '../../'
+sys.path.append(REPO_HOME_RELATIVE_PATH)
+import config as cf
+
+INCORRECT_NUM_ARGS_MESSAGE = 'invalid number of arguments: specify alignment type, subject1 and subject2'
+ILLEGAL_ALIGNMENT_ARG_MESSAGE = 'alignment argument must be either <rcds>, <linear>, <non_linear>'
+ILLEGAL_SUBJECTS_ARG_MESSAGE = 'subject arguments must be integers corresponding to subject numbers'
+
+
+def parallelize_correlation():
+    chunk_size = len(shared_subject1) // cf.NUM_PROCESSES
+    output_correlations = sm.empty(len(shared_subject1))
+    processes = [
+        Process(target=correlation,
+                args=(shared_subject1, shared_subject2, i * chunk_size, min(
+                    (i + 1) * chunk_size, len(shared_subject1)),
+                      output_correlations)) for i in xrange(cf.NUM_PROCESSES)
+    ]
+    for p in processes:
+        p.start()
+    for p in processes:
+        p.join()
+    return output_correlations
+
+
+def correlation(shared_subject1, shared_subject2, start_index, stop_index,
+                output_correlations):
+    for i in range(start_index, stop_index):
+        output_correlations[i] = stats.pearsonr(shared_subject1[i, :],
```

Comment:
Do we need to test for normality?  pearsonsr requires that "each dataset be normally distributed"